### PR TITLE
added httpClient to allow setting proxy

### PR DESCRIPTION
**Pull request by kenuiuc**

### CODE REVIEW

File: im-sdk-core/src/test/java/com/easemob/im/server/api/user/get/UserGetTest.java [hunk 1]
```diff
@@ -26,6 +32,19 @@ public void testUserGetSingle() {
...
+    public void testUserGetSingleUsingHttpClientProxy() {
+        EMProxy proxyInfo = EMProxy.builder()
+                .setIP("127.0.0.1")
+                .setPort(18888)
+                .build();
+        HttpClient httpClient = this.context.getHttpClient().block();
+        this.context.setHttpClient(Mono.just(httpClient.proxy(proxy -> proxy.type(ProxyProvider.Proxy.HTTP)
+                .address(new InetSocketAddress(proxyInfo.getIp(), proxyInfo.getPort())))));
+
+        assertThrows(RuntimeException.class, () -> this.userGet.single("username").block(Duration.ofSeconds(3)));
```
Comment (kenuiuc) [inline im-sdk-core/src/test/java/com/easemob/im/server/api/user/get/UserGetTest.java:45]: 这个测试是啥

File: im-sdk-core/src/main/java/com/easemob/im/server/EMProxy.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        }
+
+        public Builder setPassword(String password) {
+            this.password = password;
+            return this;
+        }
+
+        public EMProxy build() {
+            if (this.ip == null) {
+                throw new EMInvalidArgumentException("the IP of setting proxy cannot be empty");
```
Comment (harry-hao) [inline im-sdk-core/src/main/java/com/easemob/im/server/EMProxy.java:82]: Simply ip must not be empty.

File: im-sdk-core/src/test/java/com/easemob/im/server/api/user/get/UserGetTest.java (hunk 1 above)
Comment (harry-hao) [inline im-sdk-core/src/test/java/com/easemob/im/server/api/user/get/UserGetTest.java:45]: Should it be EMException.class ?

**Review by harry-hao — APPROVED** [review]
LGTM.

And, don't forget to provide some docs about the proxy setting in the README.md.

**Pull request merged**